### ISSUE TITLE
Installation and Documentation Issues

### ISSUE THREAD

**Issue by chewsterB** [passbolt/passbolt_api#72]
_All problems may be as a result of the poster being a moron - caution should be used at all times!_

# **Main Problem.**

Install barfs as follows

Using the install command

`su -s /bin/bash -c "app/Console/cake install --no-admin" www-data`

```
Warning Error: file_put_contents(/var/www/html/passbolt/app/Config/anonymous_statistics.php): failed to open stream: Permission denied in [/var/www/html/passbolt/app/Model/AnonymousStatistic.php, line 80]


Welcome to CakePHP v2.9.4 Console
---------------------------------------------------------------
App : app
Path: /var/www/html/passbolt/app/
---------------------------------------------------------------
 Datasource : default
 File :/var/www/html/passbolt/app/tmp/schema/default_default.sql
sh: 1: Syntax error: "(" unexpected
 Error: Something went wrong!
---------------------------------------------------------------

 Passbolt installation success! Enjoy! ☮

root@ip-x-x-x-235:/var/www/html/passbolt#
```

Ends up with the "Cake Request Collector" when trying to access the machine via a browser.

Following the final step and trying to add the user ends up with a 404 for the user link.

**Environment**

```Ubuntu 16.04 - patched at 2017-05.19

cat /proc/version
Linux version 4.4.0-1013-aws (buildd@lcy01-32) (gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.4) ) #22-Ubuntu SMP Fri Mar 31 15:41:31 UTC 2017

apache2 -v
Server version: Apache/2.4.18 (Ubuntu)
Server built:   2017-05-05T16:32:00```

php -v
PHP 7.0.15-0ubuntu0.16.04.4 (cli) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.15-0ubuntu0.16.04.4, Copyright (c) 1999-2017, by Zend Technologies

mysql-server-core-5.7            5.7.18-0ubuntu0.16.04.1                    amd64        MySQL database server binaries

```

---

## **Comments on Installation Documentation**

_As noted - the poster may be a moron - your mileage may vary!_

**Generating gnupg keys**

Somewhat confusing that the export commands go to directories that don't exist, as the code hasn't been collected yet

**Core Config**

As SSL is basically an absolute requirement here - would be really helpful to change

`App.fullBaseUrl `- to have https://<your domain here without the slash>

As otherwise I have no idea whether the http / https actually matters

**app.php config**

> You will need to specify the details of the GPG servery key, e.g. the location of the public and private key and the fingerprint. You also need to make sure that the webserver can access the gpg keyring. You can either add set it manually with $GNUPGHOME or in your config as follow:

What gpg keyring...? :) - do we mean the one I generated earlier as root that lives in /root/.gnupg...?
Do I need to copy that to /home/www-data

Then - more confusion for the terminally thick:-

**Fingerprint:** There's no clue in the documentation as to where I am getting the fingerprint from - am I getting it from the keys that I generated earlier, and if so how...?

**Location of Keys**. Presuming I've followed the instructions for exporting keys - what am I supposed to be putting here? 

Some kind of mapping to say "these fields equate to this" - would be really useful (in terms of relationship of directories to fields)

Many thanks to all concerned - I really want this software, the demo rocks and solves a bucket load of problems for us - so will be continuing to try and make this install work!

**Comment by kevinmuller** [comment]
Hi @chewsterB 

Thanks for your comments.

Regarding your installation issue, we have never seen this error before. @cedricalfonsi has a task in our JIRA to try it in a similar environment and try to reproduce it.
Did you manage to install it eventually?

Regarding your comments about the documentation, I agree with you. I have also created a task about the same to improve it (PASSBOLT-2108). I'm sure that a few new users will be happy about that !

Cheers,
Kevin

**Comment by chewsterB** [comment]
Thanks for getting in contact...

Install

A) So my problem with getting to the cake request collector (in the browser) - was my muppetry, I hadn't configured the security settings in apache for that directory - so cake was doing its job properly, because modrewrite was failing.

B) The errors at the end of the install script - well they could be related to A) - but seems like a bit of a stretch. 

Interestingly I then junked that install and then junked the server - and left the whole thing alone for a week. After that, the new release came out with Groups (which is exactly what we needed) - so reinstalled with RDS as the database and a clean EC2 instance - and boom, we have an up and running Passbolt, which is now being rolled out to the team!

In short - I would be inclined to close this ticket, its probably an anomaly and could be put down to user muppetry.

If you want me to try and recreate - then I can spin up another install and see if I can repeat the problem with the new release and a more controlled and documented install. Like most things in IT, its kind of hard to find out what the numpty is doing, until you can see the numpty actually doing what they are complaining about :)

Similarly, if you want someone to QA the install docs - then don't hesitate to let me know - Passbolt is brilliant, so more than happy to support

**Comment by jmicks8** [comment]
I am having the same issue as above. the error i am getting is:
` Datasource : default
 File :/var/www/passbolt/app/tmp/schema/default_default.sql
sh: 1: Syntax error: ")" unexpected
 Error: Something went wrong!
---------------------------------------------------------------

 Passbolt installation success! Enjoy! ☮`

Server details:
`Linux version 4.4.0-79-generic (buildd@lcy01-30) (gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.4) ) #100-Ubuntu SMP Wed May 17 19:58:14 UTC 2017

Server version: Apache/2.4.18 (Ubuntu)
Server built:   2017-06-26T11:58:04

PHP 7.0.18-0ubuntu0.16.04.1 (cli) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies with Zend OPcache v7.0.18-0ubuntu0.16.04.1, Copyright (c) 1999-2017, by Zend Technologies`

I have installed, deleted and reinstalled with the settings described in the tutorial. I am not really sure where to go from here so i appreciate any help you can provide.
